HIP/hip-336.md

```diff
@@ -0,0 +1,301 @@
+- recipient: the receiver of tokens when the spender issues a `transferFrom` operation.
+
+### Add Allowance Balances to CryptoGetInfoResponse
+
+The repeated fields `allowances` and `tokenAllowances` will be added to `CryptoGetInfoResponse` message. The `allowances` and `tokenAllowances` fields will contain hbar and token (fungible and non-fungible) allowances respectively.
+
+```protobuf
+// existing protobuf
+message CryptoGetInfoResponse {
+	// new nested message
+	message CryptoAllowance {
+		AccountID spender = 1;
+		int64 allowance = 2;
+	}
+
+	// new nested message
+	message TokenAllowance {
+		TokenID tokenId = 1;
+		AccountID spender = 2;
+		google.protobuf.Int64Value allowance = 3;
+		int32 decimals = 4;
+		google.protobuf.BoolValue approvedForAll = 5;
+	}
+	
+	// existing nested message
+	message AccountInfo {
+		// existing fields
+		AccountID accountID = 1;
+		string contractAccountID = 2;
+		bool deleted = 3;
+		AccountID proxyAccountID = 4;
+		int64 proxyReceived = 6;
+		Key key = 7;
+		uint64 balance = 8;
+		uint64 generateSendRecordThreshold = 9 [deprecated=true];
+		uint64 generateReceiveRecordThreshold = 10 [deprecated=true];
+		bool receiverSigRequired = 11;
+		Timestamp expirationTime = 12;
+		Duration autoRenewPeriod = 13;
+		repeated LiveHash liveHashes = 14;
+		repeated TokenRelationship tokenRelationships = 15;
+		string memo = 16;
+		int64 ownedNfts = 17;
+		int32 max_automatic_token_associations = 18;
+
+		// new fields
+		int32 totalAllowances = 19;
+		repeated CryptoAllowance allowances = 20;
+		repeated TokenAllowance tokenAllowances = 21;
+	}
+
+	// existing fields
+	ResponseHeader header = 1;
+	AccountInfo accountInfo = 2;
+}
+```
+
+The `TokenAllowance` message will contain the allowance information for a single fungible or non-fungible token spendable by the `spender`.
+
+If the token type is non-fungible, the `approvedForAll` field may be set to `true` if the spender has been granted access to all instances of the NFT owned by the owner. Fungible tokens will not `approvedForAll` as this field is only relevant for NFTs.
+
+The `allowances` and `approvedForAll` fields in the `TokenAllowance` message are specified using wrapper types so callers can determine whether the field has a value or not, as opposed to inferring from the type default value (for example, 0 or false).
+
+The `decimals` field, which specifies the number of decimal places the token is divisible by, is returned as a courtesy to the caller so another call to `TokenGetInfo` is not required. This field will have value `0` for NFTs.
+
+Only allowances with a non-zero value will be returned.
+
+The `spender`'s allowance will change as tokens are spent or additional allowances are approved by the token owner.
+
+The current number of allowances approved by the owner is specified in `totalAllowances`. This value is calculated as the size of `allowances` plus the size of `tokenAllowances`.
```

Comment:
What is the reason for `totalAllowances`?

Reply:
It's actually sum(size(allowances) + size(tokenAllowances)). I thought it would be easier if it was provided, but since it's so easy to derive I guess we can save the extra bytes.